src/rating/rating.ts

```diff
@@ -130,12 +135,16 @@ export class NgbRating implements ControlValueAccessor,
   ariaValueText() { return `${this.nextRate} out of ${this.max}`; }
 
   enter(value: number): void {
+    if (this.mode === 'selection') {
+      return;
+    }
+
     if (!this.readonly && !this.disabled) {
       this._updateState(value);
     }
+
     this.hover.emit(value);
```

Comment:
We still want to emit this output whenever we hover a rating. Your test should only prevent the state update.